### PR TITLE
feat: new guild creation flow

### PR DESCRIPTION
**Pull request by zgen-marci**

### CODE REVIEW

File: src/components/[guild]/Onboarding/components/FloatingProgress.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import GuildCreationProgress from "components/create-guild/GuildCreationProgress"
+import { useOnboardingContext } from "./OnboardingProvider"
+
+const FloatingProgress = ({}) => {
```
Comment (BrickheadJohnny) [inline src/components/[guild]/Onboarding/components/FloatingProgress.tsx:4]: We only use this component in one place, so we might not even need it

File: src/components/[guild]/RolePlatforms/components/AddRoleRewardModal/components/AddContractCallPanel/components/CreateNftForm/CreateNftForm.tsx [hunk 2]
```diff
@@ -145,273 +149,282 @@ const CreateNftForm = ({ onSuccess }: Props) => {
...
-          <GridItem colSpan={{ base: 3, sm: 1 }}>
-            <ImagePicker />
-          </GridItem>
-
-          <GridItem colSpan={{ base: 3, sm: 2 }}>
-            <Stack spacing={6}>
-              <HStack alignItems="start">
-                <FormControl isInvalid={!!errors?.name}>
-                  <FormLabel>Name</FormLabel>
+      <ModalHeader>Create NFT</ModalHeader>
```
Comment (BrickheadJohnny) [inline src/components/[guild]/RolePlatforms/components/AddRoleRewardModal/components/AddContractCallPanel/components/CreateNftForm/CreateNftForm.tsx:152]: This breaks the styling in the "+ Add reward" modal (guild page), so let's keep the old CreateNftForm component and just wrap it in a modal in the create guild flow.

File: src/components/common/DiscordGuildSetup/DiscordGuildSetup.tsx [hunk 3]
```diff
@@ -93,7 +94,10 @@ const DiscordGuildSetup = ({
               onSelect={
                 selectedServer
                   ? undefined
-                  : (newServerId) => setValue(fieldName, newServerId)
+                  : (newServerId) => {
```
Comment (BrickheadJohnny) [inline src/components/common/DiscordGuildSetup/DiscordGuildSetup.tsx:97]: We can revert this change and use the `useGateables` (for Discord) or `useIsTGBotIn` (for Telegram) hook inside the `BasicInfo` component in order to set the guild image from Discord server image / Telegram group image

\+ you can remove the `useSetRoleImageAndNameFromPlatformData` hook from the `TelegramGroup` & `ServerSetupCard` components, we don't need to set the image/name from there anymore

File: src/components/create-guild/Pagination.tsx [hunk 4]
```diff
@@ -60,14 +60,14 @@ const Pagination = ({
...
+        </Button>*/}
         {!nextButtonHidden && (
           <Button
-            colorScheme="indigo"
+            colorScheme="green"
             isDisabled={nextButtonDisabled}
             onClick={nextStepHandler ?? nextStep}
           >
-            {nextStepText}
+            Add{/*nextStepText*/}
```
Comment (BrickheadJohnny) [inline src/components/create-guild/Pagination.tsx:70]: Actually we don't need the Pagination component anymore, we can just use a simple Button instead as we discussed

File: src/components/create-guild/hooks/useSetImageAndNameFromPlatformData.ts [hunk 5]
```diff
@@ -23,7 +23,13 @@ const useSetImageAndNameFromPlatformData = (
 
   useEffect(() => {
     if (!onUpload) return
-    if (!(platformImage?.length > 0) || !!touchedFields.imageUrl) {
+
+    if (platformImage && platformImage.includes("/guildLogos")) {
```
Comment (BrickheadJohnny) [inline src/components/create-guild/hooks/useSetImageAndNameFromPlatformData.ts:27]: I think we can remove this extra logic once you implement the changes which I mentioned in my comment inside `DiscordGuildSetup.tsx`

File: src/components/[guild]/Onboarding/components/FloatingProgress.tsx (hunk 1 above)
Comment (zgen-marci) [reply src/components/[guild]/Onboarding/components/FloatingProgress.tsx:4]: I found the reason I used a new component here. It uses the `DynamicOnboardingProvider` context. I can't use it yet a layer above.

File: src/components/create-guild/MultiPlatformGrid/components/MultiPlatformSelectButton.tsx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+      }
+    }
+
+    return platformAddedToGuild
+  }
+
+  return (
+    <DisplayCard
+      cursor="pointer"
+      onClick={
```
Comment (BrickheadJohnny) [inline src/components/create-guild/MultiPlatformGrid/components/MultiPlatformSelectButton.tsx:100]: TODO: should be disabled if isDone is true (?) - will need to discuss it

File: src/components/create-guild/hooks/useCreateGuild.tsx [hunk 7]
```diff
@@ -34,6 +36,25 @@ const useCreateGuild = () => {
     onSuccess: (response_) => {
       triggerConfetti()
 
+      setTimeout(() => {
```
Comment (BrickheadJohnny) [inline src/components/create-guild/hooks/useCreateGuild.tsx:39]: why do we need a setTimeout here? 👀

File: src/components/create-guild/MultiPlatformGrid/components/MultiPlatformSelectButton.tsx [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+    () => onSelection(platform),
+    false,
+    "creation"
+  )
+
+  const guildPlatforms = useWatch({ name: "guildPlatforms" })
+  const twitterLink = useWatch({ name: "socialLinks.TWITTER" })
+
+  const removePlatform = (platformName: PlatformName) => {
+    const guildPlatformsCopy = JSON.parse(JSON.stringify(guildPlatforms))
```
Comment (BrickheadJohnny) [inline src/components/create-guild/MultiPlatformGrid/components/MultiPlatformSelectButton.tsx:96]: I think we don't need a deep copy of guildPlatform here, since we don't modify it

File: src/components/create-guild/templates.ts [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+      {
+        data: { minAmount: 3 },
+
+        type: "SNAPSHOT_VOTES",
+        chain: "ETHEREUM",
+      },
+      {
+        type: "RABBITHOLE",
+        address: "0x2face815247a997eaa29881c16f75fd83f4df65b",
+        chain: "ETHEREUM",
```
Comment (BrickheadJohnny) [inline src/components/create-guild/templates.ts:219]: we need to set `data` as an empty object too, since there's a bug in gate right now and the guild won't get created unless we define that

**Review by BrickheadJohnny — APPROVED** [review]
